src/main/java/org/sagebionetworks/web/client/widget/team/JoinTeamWidget.java

```diff
@@ -107,7 +112,52 @@ public void configure(String teamId, boolean isChallengeSignup, TeamMembershipSt
 		this.isMemberMessage = isMemberMessage;
 		this.successMessage = successMessage;
 		this.buttonText = buttonText;
-		view.configure(authenticationController.isLoggedIn(), teamMembershipStatus, isMemberMessage, buttonText, requestOpenInfoText, isSimpleRequestButton);
+		this.teamMembershipStatus = teamMembershipStatus;
+		view.clear();
+		if (buttonText != null && !buttonText.isEmpty()) {
+			view.setJoinButtonText(buttonText);
+		}
+		if (requestOpenInfoText != null && !requestOpenInfoText.isEmpty()) {
+			view.setRequestOpenText(requestOpenInfoText);
```

Comment:
See above:

"The default, which is set in the XML as well as by view.clear(),  sets the default, so don't touch the default unless there is an alternative."